File: android-aop-core/src/main/java/com/flyjingfish/android_aop_core/annotations/MainThread.java
package com.flyjingfish.android_aop_core.annotations;

import com.flyjingfish.android_aop_annotation.anno.AndroidAopPointCut;
import com.flyjingfish.android_aop_core.cut.MainThreadCut;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 切换到主线程的操作，加入此注解可使你的方法内的代码切换到主线程执行
 */
@AndroidAopPointCut(MainThreadCut.class)
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface MainThread {
}

File: android-aop-extra/src/main/java/com/flyjingfish/android_aop_core/annotations/MainThread.kt
package com.flyjingfish.android_aop_core.annotations

import com.flyjingfish.android_aop_annotation.anno.AndroidAopPointCut
import com.flyjingfish.android_aop_core.cut.MainThreadCut

/**
 * 切换到主线程的操作，加入此注解可使你的方法内的代码切换到主线程执行
 */
@AndroidAopPointCut(MainThreadCut::class)
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER
)
annotation class MainThread
